### PR TITLE
Handle the browsers which don't support promises (eg. Edge)

### PR DESCRIPTION
**Pull request by kai918**
Compatibility with environments that have the browser namespace but do not support promises.

I was inspired by Snoak's answer in the following issue: mozilla#3

### CODE REVIEW

**Comment by Croydon** [comment]
What's up with this pull request?

**Comment by mi-g** [comment]
This change is dedicated to handling browsers (Edge) that offer a `browser` global but do not provide a native promise return in WebExtensions methods.
The detection of this case is done by calling a method that requires a callback (necessary because no promise returns). If the method generates an exception, it means it does not support promise and the `webextension-polyfill` code must do its job.
Unfortunately in this PR, the method chosen to test the promise availability is `browser.runtime.getPlatformInfo()` which indeed generates an exception on Edge, but primarily because this method is not implemented in this browser ([see MDN documentation](https://developer.mozilla.org/en-US/Add-ons/WebExtensions/API/runtime/getPlatformInfo)).
Using another method, like `browser.runtime.getBackgroundPage()` to test the promise availability would be a better idea.

File: src/browser-polyfill.js [hunk 1]
```diff
@@ -6,7 +6,16 @@
...
 
-if (typeof browser === "undefined") {
+var _supportsPromises = false;
+try {
+  _supportsPromises = browser.runtime.getPlatformInfo() instanceof Promise;
+} catch (e) {
+  console.log("promises not supported.");
+}
+
+if (typeof browser === "undefined" || !_supportsPromises) {
```
Comment (ExE-Boss) [inline src/browser-polyfill.js]: A better option would be to change this line to:
```javascript
if (typeof browser === "undefined" || !(() => {
  let supportsPromises = false;
  try {
    supportsPromises = browser.runtime.getPlatformInfo() instanceof Promise;
  } catch (e) {
    console.log("promises not supported.");
  }
  return supportsPromises;
})()) {
```
And remove the previous block.

**Comment by rpl** [comment]
Closing in favor of #114 (which is still exploring how we may allow the polyfill to work on Edge, and it is also taking advantage of the recently added integration tests to ensure that the additional changes needed are not going to break the polyfill behaviors expected on Chrome and Firefox).

**Comment by lfjr** [comment]
So do I need to do anything to fix it other than the tools that I'm downloading

**Pull request closed without merging**